### PR TITLE
feat: progress circle add support gradient

### LINKED ISSUE #17286: Progress type='circle' 不能设置strokeColor 为渐变色
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/embed/small-snow-j6ij6)

### Steps to reproduce
https://ant.design/components/progress-cn/#API
image

### What is expected?
Progress type='circle' 能设置strokeColor 为渐变色

### What is actually happening?
Progress type='circle' 不能设置strokeColor 为渐变色

| Environment | Info |
|---|---|
| antd | 3.19.8 |
| React | latest |
| System | windows10 |
| Browser | chrome74 |

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Solution
close #17286

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Progress.Circle  support gradient line color|
| 🇨🇳 Chinese |Progress.Circle 支持渐变色|

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/progress/demo/gradient-circle.md](https://github.com/hengkx/ant-design/blob/feature-circle-gradient/components/progress/demo/gradient-circle.md)
[View rendered components/progress/index.en-US.md](https://github.com/hengkx/ant-design/blob/feature-circle-gradient/components/progress/index.en-US.md)
[View rendered components/progress/index.zh-CN.md](https://github.com/hengkx/ant-design/blob/feature-circle-gradient/components/progress/index.zh-CN.md)

### CODE REVIEW

File: components/progress/demo/gradient-circle.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      strokeColor={{
+        '0%': '#108ee9',
+        '100%': '#87d068',
+      }}
+      percent={90}
+    />
+  </div>
+);
+
+ReactDOM.render(<Demo />, mountNode);
```
Comment (afc163) [inline components/progress/demo/gradient-circle.md]: 可以合并到已有的渐变色那个例子里去。

**Comment by afc163** [comment]
预期的 changelog 麻烦帮忙写一下。

**Comment by afc163** [comment]
ts 定义也需要更新一下。

**Comment by hengkx** [comment]
> ts 定义也需要更新一下。

这个ts定义和之前的是一样的，changelog 我改下

**Comment by hengkx** [comment]
image

File: components/progress/demo/gradient-circle.md (hunk 1 above)
Comment (hengkx) [reply components/progress/demo/gradient-circle.md]: removed

**Comment by afc163** [comment]
高大上！

**Review by afc163 — APPROVED** [review]

**Pull request merged**